### DISCUSSION TITLE
"Content String Enrichment" in "Network Request" question

### DISCUSSION THREAD

**Discussion by kurogedelic** [RavenSystem/esp-homekit-devices#2141] [MEPLHAA Scripts]
Hello.
I want to use the Network Request for my PHP, but I am having trouble with the expansion of `#HAA@aacc`'. I am writing it in the URL, not in the content.
My PHP that logs temperature and humidity sensor data sent via URL GET queries.
The following cURL command succeeds in doing so.
So, is it possible that the expansion of #HAA@aacc is only happening within the 'contents'?

cURL command (success) :
`http://host.com/data.php?write=true&key=xxxxxtemp2=25.0&hum2=99`

MEPLHAA Script (Nothing happens):
```
.....
"a": [
        {
            "t": 24,
            "n": 2,
            "g": 4,
            "k": 10,
            "j": 30,
            "y0": [
                {
                    "v": 0,
                    "r": 1,
                    "0": {
                        "h": [
                            {
                              "h": "host.com",
                              "u": "data.php?write=true&key=xxxxx&temp2=#HAA@0000&hum2=#HAA@0001"
                            }
                          ]
                    }
                }
            ]
        },
        .....
 ]
```

Thanks!

**Comment by paulzhdanov** [comment]
This is not documented, but there are 2 major issues in HAA:
1. you must use POST method to send parameter
2. you must send specific header
Here's the example which works for me:

```
        "h": [{
          "h": "192.168.1.1",
          "m": 2,
          "e": "Content-type: application/x-www-form-urlencoded\r\n",
          "u": "write_data.php",
          "c": "t1=#HAA@0100&t2=#HAA@0200&t3=#HAA@0300"
          }] 

```
hope that helps

**Reply by kurogedelic** [reply]
I tried changing it like that, but it doesn't seem to work.

```
         "h": [
                            {
                              "h": "host.com",
                              "m": 2,
                              "e": "Content-type: application/x-www-form-urlencoded\r\n",
                              "u": "data.php",
                              "c": "write=true&key=xxxx&temp2=#HAA@0100&hum2=#HAA@0200"
                            }
                          ]
```

However, using the code I posted at the very beginning,
If `#HAA@aabb` is not used, the data is sent to PHP.
So I think `#HAA@aabb` is not converted to a string well.
What does `aabb` refer to? 

Thanks.

**Reply by paulzhdanov** [reply]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Accessory-Configuration#content-string-enrichment

`If the string contains the magic expression #HAA@aacc then the expression will be replaced by the characteristic cc value from the service aa e.g. #HAA@0100 will be replaced with the value held by characteristic 0 from service 1.
`

**Reply by paulzhdanov** [reply]
debug the log with "o":4 option in config, it should help

**Comment by kurogedelic** [comment]
If connect with serial and look at the serial port,Found this:
`! <1> TCP (-3)`

I set it to` "m":2` and also changed PHP to receive POST.
This is an error, isn't it?
